### PR TITLE
Replace PaddlePaddle ResNet50 example with one from the DeepLearningExamples

### PR DESCRIPTION
**Pull request by JanuszL**
- replaces PaddlePaddle ResNet50 example with one from the DeepLearningExamples
  which uses the most recent PaddlePaddle API

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- replaces PaddlePaddle ResNet50 example with one from the DeepLearningExamples
  which uses the most recent PaddlePaddle API

### Affected modules and functionalities:
- ResNet50 PaddlePaddle example

### Key points relevant for the review:
- readme
- mostly copy-past from DL example, with some trimming and additional fixes for the latest version of PaddlePaddle

### Tests:

- [x] Existing tests apply
  - TL3_RN50_convergence
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: docs/examples/use_cases/paddle/resnet50/paddle-resnet50.rst [hunk 1]
```diff
@@ -1,49 +1,48 @@
...
-.. code-block:: bash
-
-   python -m paddle.distributed.launch --selected_gpus 0,1,2,3,4,5,6,7 main.py -d 50 [imagenet-folder with train and val folders]
-
-The training schedule in `He et al. 2015 <https://arxiv.org/abs/1512.03385>`_ was used where learning rate starts at 0.1 and decays by a factor of 10 every 30 epochs.
 
 Usage
 -----
 
+Before using install the necessary packages from requirements.txt.
```
Comment (mzient) [inline docs/examples/use_cases/paddle/resnet50/paddle-resnet50.rst:27]:
```suggestion
Before running the example install the necessary packages from requirements.txt.
```
or simply
```suggestion
Install the necessary packages from requirements.txt before use.
```

**Review by mzient — APPROVED** [review]

File: docs/examples/use_cases/paddle/resnet50/paddle-resnet50.rst (hunk 1 above)
Comment (JanuszL) [reply docs/examples/use_cases/paddle/resnet50/paddle-resnet50.rst:27]: Done

File: docs/examples/use_cases/paddle/resnet50/paddle-resnet50.rst [hunk 2]
```diff
@@ -1,49 +1,48 @@
...
+* For training:
+  * Normalization
+  * Random resized crop to 224x224
+    * Scale from 8% to 100%
+    * Aspect ratio from 3/4 to 4/3
+  * Random horizontal flip
+* For inference:
+  * Normalization
+  * Scale to 256x256
+  * Center crop to 224x224
```
Comment (stiepan) [inline docs/examples/use_cases/paddle/resnet50/paddle-resnet50.rst:21]: It looks like the middle tier of the list is not parsed as such.

image

Comment (JanuszL) [reply docs/examples/use_cases/paddle/resnet50/paddle-resnet50.rst]: Maybe now

**Comment by JanuszL** [comment]
!build

File: docs/examples/use_cases/paddle/resnet50/dali.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    std: float
+    output_dtype: types.DALIDataType
+    output_layout: str
+    pad_output: bool
+
+def create_dali_pipeline(batch_size, num_threads, device_id, data_dir, ops_meta,
+                         shard_id, num_shards, dali_cpu=False, is_training=True,
+                         seed=12):
+    pipeline = Pipeline(batch_size, num_threads, device_id, seed=seed)
+    with pipeline:
```
Comment (stiepan) [inline docs/examples/use_cases/paddle/resnet50/dali.py:48]: It's a bit of a nitpick, but I wonder if we could just go here for pipeline_def decorator? It feels a bit like a vestige of class-based API (that would, for example, prevent us from running auto_augments).

```suggestion
@pipeline_def(seed=12)
def dali_pipeline(data_dir, ops_meta, shard_id, num_shards, dali_cpu=False, is_training=True):
```

Comment (JanuszL) [reply docs/examples/use_cases/paddle/resnet50/dali.py]: Done

File: docs/examples/use_cases/paddle/resnet50/train.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                self.metric_dict[key] = new_metrics[key]
+
+    @property
+    def is_updated(self):
+        return self.metric_dict is not None
+
+
+def main(args):
+    """
+    A enterpoint to train and evaluate a ResNet50 model, which contains six steps.
```
Comment (stiepan) [inline docs/examples/use_cases/paddle/resnet50/train.py:68]:
```suggestion
    An entrypoint to train and evaluate a ResNet50 model, which contains six steps.
```

Comment (JanuszL) [reply docs/examples/use_cases/paddle/resnet50/train.py:68]: Done

File: docs/examples/use_cases/paddle/resnet50/utils/affinity.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    Generate three dict objects, gpu_cpu_affinity_map, cpu_socket_gpus_list, cpu_core_groups.
+    gpu_cpu_affinity_map (dict): Key is GPU ID and value is cpu_affinity string.
+    cpu_socket_gpus_list (dict): Key is cpu_affinity string and value is  a list
+                                 collected all GPU IDs that affinity to this cpu socket.
+    cpu_core_groups (dict):      Key is cpu_affinity string and value is cpu core groups.
+                                 cpu core groups contains #GPUs groups, each group have,
+                                 nearly eaual amount of cpu cores.
+
+    Example:
+        $nvidis-smi topo -m
```
Comment (stiepan) [inline docs/examples/use_cases/paddle/resnet50/utils/affinity.py:31]:
```suggestion
        $nvidia-smi topo -m
```

File: docs/examples/use_cases/paddle/resnet50/utils/affinity.py [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    local_rank = paddle.distributed.get_rank() % gpu_per_node
+
+    # gpu_cpu_affinity_map (dict): Key is GPU ID and value is cpu_affinity string.
+    # cpu_socket_gpus_list (dict): Key is cpu_affinity string and value is  a list
+    #                              collected all GPU IDs that affinity to this cpu socket.
+    # cpu_core_groups (dict):      Key is cpu_affinity string and value is cpu core groups.
+    #                              cpu core groups contains #GPUs groups, each group have,
+    #                              nearly eaual amount of cpu cores.
+    # Example:
+    # $nvidis-smi topo -m
```
Comment (stiepan) [inline docs/examples/use_cases/paddle/resnet50/utils/affinity.py:174]:
```suggestion
    # $nvidia-smi topo -m
```

**Review by stiepan — APPROVED** [review]
Just a two more typos.

File: docs/examples/use_cases/paddle/resnet50/utils/affinity.py (hunk 5 above)
Comment (JanuszL) [reply docs/examples/use_cases/paddle/resnet50/utils/affinity.py]: Done

**Review by stiepan — APPROVED** [review]

**Review by awolant — APPROVED** [review]

**Pull request merged**